### PR TITLE
[enhance] Process morphological mode in explore graph paths

### PR DESCRIPTION
**Pull request by Eradan94**
## Description

Process the morphological mode in explore graph paths.

## Motivation and Context

Issue #80 

## How Has This Been Tested?

Tested with several graphs in French, Malagasy and Korean
Malagasy : V-tous.grf
French : see attachment and Dnum-ord.grf
Korean : tested with graphs given by Pr.Nam, but unsure about the jamo processing
[tested-graphs.zip](https://github.com/UnitexGramLab/unitex-core/files/3320190/tested-graphs.zip)

## Type of files

- [ ] `bin`: Binary files
- [ ] `ci`: Continuous integration files
- [ ] `doc`: Documentation files
- [x] Plain-text source code files

## Level of change

- [ ] `break`: Breaking change
- [ ] `exp`: Experimental change
- [ ] `tmp`: Temporal change
- [ ] `major`: Major change
- [ ] `minor`: Minor change
- [ ] `sec`: Vulnerability-related change
- [x] None of the above (normal change)

## Type of change

- [ ] `deprecat`: Deprecation of a once-stable feature
- [x] `enhance`: Enhancement in existing functionality
- [ ] `fix`: Bug fix
- [ ] `feature`: New feature
- [ ] `hotfix`: Hotfix for bugs
- [ ] `refactor`: Improve coding style, comments
- [ ] `remove`: Remove a feature

## Checklist:

- [x] My code compiles
- [x] My code follows the code style of this project
- [ ] My code includes javadoc/doxygen where appropriate
- [x] My code is well factored, so that there is not repetitive code in the wild
- [x] My code does not require a change in the documentation, if so I already opened an issue to list the changes
- [x] I have read the **CONTRIBUTING** document
- [x] I have read the **Commit Message Guidelines**
- [x] I understand that all commits on my pull request will be squashed to a single good one

### CODE REVIEW

File: Fst2List.cpp [hunk 1]
```diff
@@ -269,6 +269,7 @@ class CFstApp {
   char ofExt[16];
   char ofnameOnly[512];        // output file name
   char defaultIgnoreName[512]; // input file name
+  bool modeMorph = false;  // true if the current state is in morphological mode
```
Comment (martinec) [inline Fst2List.cpp]: Have you noticed this suggestion https://github.com/UnitexGramLab/unitex-core/pull/82#discussion_r297087253 ?

**Comment by martinec** [comment]
I'm closing this PR. To go forward, update instead https://github.com/UnitexGramLab/unitex-core/pull/82

File: Fst2List.cpp (hunk 1 above)
Comment (Eradan94) [reply Fst2List.cpp]: Oops, i forgot this one. Done.

File: Fst2List.cpp [hunk 2]
```diff
@@ -772,6 +773,23 @@ class CFstApp {
     inputPtrCnt = outputPtrCnt = inBufferCnt = outBufferCnt = 0;
   }
 
+  /**
+    * print one space when the morphological mode is off
+  **/
+  void outOneSpace() {
```
Comment (martinec) [inline Fst2List.cpp]: What about https://github.com/UnitexGramLab/unitex-core/pull/82/files#r297094591

Comment (Eradan94) [reply Fst2List.cpp]: I think I misunderstood your comment. What changes should I make?

Comment (martinec) [reply Fst2List.cpp]: What about `appendSingleSpace` instead of `outOneSpace` ?

Comment (Eradan94) [reply Fst2List.cpp]: It seems better than outOneSpace

**Comment by martinec** [comment]
LGTM

**Pull request merged**